reasoners/base.py

```diff
@@ -326,4 +327,18 @@ def evaluate_sc(self,
     def eval_output(self, answer, output):
         pass
 
-    
\ No newline at end of file
+class Tool():
+    def __init__(self, func, name, description):
+        self.func = func
+        self.name = name
+        self.description = description
+        self.call_params = inspect.signature(func).parameters
+
+    def __call__(self, **kwargs):
+        for param in self.call_params:
+            if param not in kwargs:
+                raise ValueError(f"Missing required parameter: {param}")
+        try:
```

Comment:
Do not use try-except in abstract class. It will hide the true error from the implementation.